models/model.js

```diff
@@ -6,7 +6,26 @@ exports.fetchTopics = () => {
         `SELECT * FROM topics;`
       )
       .then((result) => {
-        // console.log(result.rows);
         return result.rows;
       })
+};
+
+
+exports.selectArticleById = (article_id) => {
+  return db
+  .query( 'SELECT * FROM articles WHERE article_id = $1;',[article_id])
+
+  .then(({ rows }) => {
+    const article = rows[0];
+    if(!article) {
+      return Promise.reject({
+        status: 404,
+        msg: 'Page not found',
+      })
+    }
+    return article;
+  })
+  .catch((err) => {
+    return Promise.reject(err);
```

Comment:
Because you have a catch in your controller you shouldn't need a catch here, if you left it out any errors thrown would fall into your controller catch and be handled there. Get rid of this catch block

Reply:
Fixed it. Thanks